core/units.h

```diff
@@ -253,6 +253,21 @@ static inline int mbar_to_PSI(int mbar)
 	return to_PSI(p);
 }
 
+static inline int32_t altitude_to_pressure(int altitude) 	// altitude in meters above sea level (range -500 to 4999)
```

Comment:
Sorry, I am very late to the party, but: In the planner, we already have functionality to enter surface pressure either as an actual pressure or as an altitude (assuming standard pressure at sea level), see dive planner.cpp around lines 225ff. Why do it differently here? If this is superior, you should change the behaviour in the planner as well

Reply:
For those not familiar with the planner: Could you state which function transforms altitude to pressure? I only see `depth_to_atm`, but that is under depth water, isn't it?

Reply:
Of course I did not think at all of the implementation in the planner (and I use the planner regularly!). The philosophy behind my design of the converter was that it should not take more screen space than the QString that was always displayed there (i.e. the non-editable version). The implementation in the planner takes double the space but it more user friendly. With my implementation the conversion is dependent on the user selecting the unit (mbar/m) first before typing the number. The conversion is triggered when the user types Enter after typing the number to be converted. The user can type in the number, then select the unit but selecting unit does not do the conversion: click on the number and hit Enter will do it.

Reply:
> For those not familiar with the planner: Could you state which function transforms altitude to pressure? I only see `depth_to_atm`, but that is depth under water, isn't it?

It doesn't have its own function but it is done here:
https://github.com/Subsurface-divelog/subsurface/blob/14a763a6a0148a42fad06717bd2532181a54c626/desktop-widgets/diveplanner.cpp#L252
and in reverse here:
https://github.com/Subsurface-divelog/subsurface/blob/14a763a6a0148a42fad06717bd2532181a54c626/desktop-widgets/diveplanner.cpp#L238

Reply:
I wasn't so much thinking about the UI presentation but more about the computation.

Reply:
> It doesn't have its own function but it is done here

Thanks. Indeed, both places should use the same global function.

Reply:
I have no particular allegiance to the database from where I do my calculations. In this case the physicist amongst us can only have by far the best opinion.

Reply:
Robert, you have probably seen the inline function in units.h. I think you are in by far the best position to make a recommendation about the calculation. I am fairly certain your calculation is from first principles and I am quite positive towards such an approach. ok?

Reply:
Have a look at 
![image](https://user-images.githubusercontent.com/863957/57193948-1eb13c80-6f41-11e9-9523-1bd87be8139c.png) The dots are your datapoint, the solid line is the formula from diveplanner.cpp line 252 (barometric formula like https://en.wikipedia.org/wiki/Barometric_formula ). Looks good enough for me.

Reply:
Note that my formula is for altitudes in mm (as all internal length values in Subsurface). I guess you should use mm as well and only convert them to other units when displaying to the user (using the usual helper functions for depth).

Reply:
Using mm and helpers ok. If you were to look at the existing two ways of doing the calculation, which would be your preferred? I suggest using your preferred, which I will write as an inline function in units.h and using it in the Information tab as well as in the planner.

Reply:
My personal taste is always in favour of a formula instead of a look-up table. And yes, please make this a helper function (as well as the reverse) and use that everywhere.